### DISCUSSION TITLE
Saving a std::vector<Real> Material for postprocessing (excluding visualization)

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#22960] [Q&A General] [answered]
Dear all,
I am wondering whether it would be possible to store a material that contains a varying in size (on run time) std::vector of Real entries for postprocessing. I see that for  RealVectorValue material the way would be to store its entries in an auxvariable (I guess for visualization) and then retrieved them via a vectorpostprocessor. 
Since I do not need to visualize my vectors (dimensions are usually greater than 3), I was thinking to directly sample all entries in a dedicated vectorpostprocessor, something along the line of ElementVariableVectorPostprocessor though working on material properties ( this would then require to extend the SampleBase class I guess). 
The other option would be to store the list of entries into an auxvariable and then go via the ElementVariableVectorPostprocessor (though I see problems with visualization here). 
I would appreciate any feedback on my thoughts (like whether this would be doable at all).
Thanks in advance,
mauro

**Comment by GiudGiud** [comment, marked as answer]
Hello

Unfortunately, as the docs for the MaterialVPP say, it hasnt been written for array material properties, only for Real-valued ones. This would not be very hard to do, and we would welcome the contribution to the framework. 

If you don't want that, you can try this hacky way, I have not tried it but I think it would work:
You specify output = 'exodus' in the material block that defines this array material property, then looking at the exodus you can see that a number of auxiliary variables got added to the aux system. 
You use the names of these auxiliary variables in an ElementValueSampler to output them to CSV
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html

Guillaume

**Comment by mcacace** [comment]
@GiudGiud thanks for your feedback. About your first point, would you suggest to augment the MaterialVectorPostprocessor class? Will browse through that and see how that can be done. Thanks again

**Reply by GiudGiud** [reply]
I think this would make the most sense
